### PR TITLE
Code cleanup: rename l_build_ & r_max_degree & b_backtrack, remove TODOs, add CLion build instructions

### PR DESCRIPTION
**Pull request by jparismorgan**
### What
Some small code cleanups:
* Rename l_build_ & r_max_degree & b_backtrack to use lowercase snake case
* Remove TODOs that we do not need to do
* Add CLion build instructions

### Testing
CI passes

### CODE REVIEW

File: apis/python/test/test_index.py [hunk 1]
```diff
@@ -350,20 +350,6 @@ def test_index_with_incorrect_num_of_query_columns_complex(tmp_path):
                     with pytest.raises(TypeError):
                         index.query(query, k=1)
 
-                # TODO(paris): This will throw with the following error. Fix and re-enable, then remove
```
Comment (jparismorgan) [inline apis/python/test/test_index.py:353]: This was added by Andrew accidentally in a merge conflict: https://github.com/TileDB-Inc/TileDB-Vector-Search/commit/ebcf35b94377bd07b93cfd0a715043938e636364#diff-d6acf5a535e6bc7ca26f487ae25cecf98fe5244e93d1f44367a7a6bbef5ef91eR183

File: apis/python/src/tiledb/vector_search/ivf_flat_index.py [hunk 2]
```diff
@@ -466,7 +466,6 @@ def create(
         group_exists=group_exists,
         config=config,
     )
-    # TODO(paris): Save training_source_uri as metadata so that we use it for re-ingestion's.
```
Comment (jparismorgan) [inline apis/python/src/tiledb/vector_search/ivf_flat_index.py:469]: We ended up going with a different design (i.e. `copy_centroids_uri`), so do not need this.

File: apis/python/src/tiledb/vector_search/ingestion.py [hunk 3]
```diff
@@ -593,7 +593,6 @@ def create_partial_write_array_group(
             ids_array_dom = tiledb.Domain(ids_array_rows_dim)
             ids_attr = tiledb.Attr(
                 name="values",
-                # TODO(paris): Enable configuring this so Vamana will work with different ID types.
```
Comment (jparismorgan) [inline apis/python/src/tiledb/vector_search/ingestion.py:596]: We'll need to change this when we support different ID types, but it's not just specific to Vamana, so removing.

**Review by NikolaosPapailiou — APPROVED** [review]
Thanks for the cleanup!

**Pull request merged**